core/src/smartcontracts/isi/asset.rs

```diff
@@ -222,13 +222,12 @@ pub mod isi {
                 *quantity = quantity
                     .checked_add(mint.object)
                     .ok_or(MathError::Overflow)?;
-                *quantity
+                mint.object
             };
 
-            // TODO: add only value actually being minted (https://github.com/hyperledger/iroha/issues/3543)
             #[allow(clippy::float_arithmetic)]
             {
-                wsv.metric_tx_amounts += quantity.into_metric();
+                wsv.metric_tx_amounts += new_quantity.into_metric();
```

Comment:
maybe we can then use just `mint.object` directly?

Reply:
Maybe, but I think it's slightly easier to understand like this.